tests/components/azure_data_explorer/test_init.py

```diff
@@ -0,0 +1,280 @@
+                FilterTest("sensor.included_test", 1),
+                FilterTest("binary_sensor.included", 1),
+                FilterTest("binary_sensor.excluded", 0),
+            ],
+        ),
+        (
+            {
+                "exclude_domains": ["climate"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+            [
+                FilterTest("climate.excluded", 0),
+                FilterTest("light.included", 1),
+                FilterTest("sensor.excluded_test", 0),
+                FilterTest("sensor.included_test", 1),
+                FilterTest("binary_sensor.included", 1),
+                FilterTest("binary_sensor.excluded", 0),
+            ],
+        ),
+        (
+            {
+                "include_domains": ["light"],
+                "include_entity_globs": ["*.included_*"],
+                "exclude_domains": ["climate"],
+                "exclude_entity_globs": ["*.excluded_*"],
+                "exclude_entities": ["light.excluded"],
+            },
+            [
+                FilterTest("light.included", 1),
+                FilterTest("light.excluded_test", 0),
+                FilterTest("light.excluded", 0),
+                FilterTest("sensor.included_test", 1),
+                FilterTest("climate.included_test", 1),
+            ],
+        ),
+        (
+            {
+                "include_entities": ["climate.included", "sensor.excluded_test"],
+                "exclude_domains": ["climate"],
+                "exclude_entity_globs": ["*.excluded_*"],
+                "exclude_entities": ["light.excluded"],
+            },
+            [
+                FilterTest("climate.excluded", 0),
+                FilterTest("climate.included", 1),
+                FilterTest("switch.excluded_test", 0),
+                FilterTest("sensor.excluded_test", 1),
+                FilterTest("light.excluded", 0),
+                FilterTest("light.included", 1),
+            ],
+        ),
+    ],
+    ids=["allowlist", "denylist", "filtered_allowlist", "filtered_denylist"],
+)
+async def test_filter(
+    hass,
+    entry_managed,
+    tests,
+    mock_azure_data_explorer_ManagedStreamingIngestClient_ingest_data,
+) -> None:
+    """Test different filters.
+
+    Filter_schema is also a fixture which is replaced by the filter_schema
+    in the parametrize and added to the entry fixture.
+    """
+    count = 0
+
+    for test in tests:
+        count += test.expected_count
+
+        hass.states.async_set(test.entity_id, STATE_ON)
+        async_fire_time_changed(
+            hass,
+            utcnow() + timedelta(seconds=entry_managed.options[CONF_SEND_INTERVAL]),
+        )
+        await hass.async_block_till_done()
+        assert (
+            mock_azure_data_explorer_ManagedStreamingIngestClient_ingest_data.call_count
+            == count
+        )
```

Comment:
Instead of asserting the call_count, can we assert the call_args
This makes the test more robust

Reply:
Please specify what you mean, both the Call_args and the Call_count is part of the test Pytest suite

Reply:
Please point me to the line where this test is asserting the `call_args` of `mock_azure_data_explorer_ManagedStreamingIngestClient_ingest_data`

Reply:
Sorry, I am not asserting call_args, only args_count. I will be happy to change to using the args, but 1. I dont know how and 2. I cant see how it will improve the test?